hack/feature-deploy.sh

```diff
@@ -53,6 +53,16 @@ do
       continue
     fi
 
+    pre_deploy=feature-configs/${FEATURES_ENVIRONMENT}/${feature}/pre_deploy.sh
+    if [[ -f $pre_deploy ]]; then
+        echo "[INFO] Pre-deploying feature '$feature' for environment '$FEATURES_ENVIRONMENT'"
+        if ! $pre_deploy; then
+          echo "[WARN] Pre-deploy of feature '$feature' failed."
+          feature_failed=1
+          continue
+      fi
+    fi
+
```

Comment:
Does this change affect only the CI runs on changes under cnf-features-deploy/ztp, or are the CI runs for cnf-tests also affected by this change?

Reply:
It only affects the CI on changes under cnf-features-deploy/ztp right now, but the change is generic which can be leveraged by other CIs for cnf-tests who want to run something prior to feature deploy.